compaction/task_manager_module.cc

```diff
@@ -7,6 +7,33 @@
  */
 
 #include "compaction/task_manager_module.hh"
+#include "replica/database.hh"
+
+namespace std {
+
+std::ostream& operator<<(std::ostream& os, const api::table_info& ti) {
+    return os << "table{name=" << ti.name << ", id=" << ti.id << "}";
+}
+
+} // namespace std
+
+future<> run_on_existing_tables(sstring op, replica::database& db, std::string_view keyspace, const std::vector<api::table_info> local_tables, std::function<future<> (replica::table&)> func) {
+    std::exception_ptr ex;
+    for (const auto& ti : local_tables) {
+        tasks::tmlogger.debug("Starting {} on {}.{}", op, keyspace, ti);
+        try {
+            co_await func(db.find_column_family(ti.id));
+        } catch (const replica::no_such_column_family& e) {
+            tasks::tmlogger.warn("Skipping {} of {}.{}: {}", op, keyspace, ti, e.what());
+        } catch (...) {
+            ex = std::current_exception();
+            tasks::tmlogger.error("Failed {} of {}.{}: {}", op, keyspace, ti, ex);
+        }
+        if (ex) {
+            co_await coroutine::return_exception_ptr(std::move(ex));
+        }
+    }
+}
```

Comment:
this is a generic function that can be defined in api/storage_service.cc
or, we can create api/utils.{cc,hh} for generic useful stuff

Reply:
Can't we move that out of api/? Eventually it will be used only in compaction/task_manager_module.cc.

If it stays in api/ then:
- we cannot keep definition in source file as api is not compiled for tests
- keeping definition in api/utils.hh (or any header) will make us include `replica/database.hh` which is heavy and should probably be avoided

I don't really know where is the right place for that.

Reply:
utils/api.{cc,hh} can host the generic stuff used by the api layer so it can be shared with other parts of scylla.
but I don't want such generic stuff to be buried here since it may very well be reused in other rest api calls.

Reply:
looks like it was later agreed that it could live in compaction/. never mind then.